improver/calibration/rainforest_calibration.py

```diff
@@ -0,0 +1,185 @@
+DEFAULT_OUTPUT_REALIZATIONS_COUNT = 100
+
+
+class ApplyRainForestsCalibration(PostProcessingPlugin):
+    """Class to calibrate input forecast given a series of RainForests tree models."""
+
+    def __init__(self, model_config_dict: dict, threads: int):
+        """Initialise the tree model variables used in the application of RainForests
+        Calibration.
+
+        Args:
+            model_config_dict:
+                Dictionary containing Rainforests model configuration variables.
+            threads:
+                Number of threads to use during prediction with tree-model objects.
+
+        Dictionary is of format::
+
+            {
+                "-50.0" : {
+                    "lightgbm_model" : "<path_to_lightgbm_model_object>",
+                    "treelite_model" : "<path_to_treelite_model_object>"
+                },
+                "-25.0" : {
+                    "lightgbm_model" : "<path_to_lightgbm_model_object>",
+                    "treelite_model" : "<path_to_treelite_model_object>"
+                },
+                ...,
+                "50.0" : {
+                    "lightgbm_model" : "<path_to_lightgbm_model_object>",
+                    "treelite_model" : "<path_to_treelite_model_object>"
+                }
+            }
+
+        The keys specify the error threshold value, while the associated values
+        are the path to the corresponding tree-model objects for that threshold.
+
+        Treelite predictors are used if treelite_runitme is an installed dependency
+        and an associated path has been provided for all thresholds, otherwise lightgbm
+        Boosters are used as the default tree model type.
+        """
+        from lightgbm import Booster
+
+        try:
+            from treelite_runtime import Predictor
+
+            self.treelite_enabled = True
+        except ModuleNotFoundError:
+            warnings.warn(
+                "Module treelite_runtime unavailable. Defaulting to using lightgbm Boosters."
+            )
+            self.treelite_enabled = False
+
+        # Dictionary keys represent error thresholds, however may be strings as they
+        # are sourced from json files. In order use these in processing, and to sort
+        # them in a sensible fashion, we shall cast the key values as float32.
+        model_config_dict = {
+            np.float32(threshold): model_config_dict[threshold]
+            for threshold in model_config_dict.keys()
+        }
+        sorted_model_config_dict = OrderedDict(sorted(model_config_dict.items()))
+
+        self.error_thresholds = np.array([*sorted_model_config_dict.keys()])
+
+        lightgbm_model_filenames = [
+            threshold_dict.get("lightgbm_model")
+            for threshold_dict in model_config_dict.values()
+        ]
+        treelite_model_filenames = [
+            threshold_dict.get("treelite_model")
+            for threshold_dict in model_config_dict.values()
+        ]
```

Comment:
```suggestion
        sorted_model_config_dict = OrderedDict(
            sorted({np.float32(k): v for k, v in model_config_dict.items()}.items())
        )

        self.error_thresholds = np.array([*sorted_model_config_dict.keys()])

        lightgbm_model_filenames = [
            threshold_dict.get("lightgbm_model")
            for threshold_dict in sorted_model_config_dict.values()
        ]
        treelite_model_filenames = [
            threshold_dict.get("treelite_model")
            for threshold_dict in sorted_model_config_dict.values()
        ]
```

Reply:
Thanks for this suggestion @MoseleyS. I've gone with the above and expanded the unit test `test__init__for_ordered_error_threshold` to check that the sorting is as expected and that the consistency between error threshold and tree_model is preserved.